boards/hifive1/include/periph_conf.h

```diff
@@ -82,6 +80,24 @@ extern "C" {
 #define PWM_NUMOF                   (3)
 /** @} */
 
+/**
+ * @name    SPI device configuration
+ *
+ * @{
+ */
+/* DIV_UP is division which rounds up instead of down */
+#define DIV_UP(a,b) (((a) + ((b) - 1)) / (b))
+static const uint32_t spi_clk_config[] = {
+    DIV_UP(CLOCK_CORECLOCK, 2 *   100000) - 1,
+    DIV_UP(CLOCK_CORECLOCK, 2 *   400000) - 1,
+    DIV_UP(CLOCK_CORECLOCK, 2 *  1000000) - 1,
+    DIV_UP(CLOCK_CORECLOCK, 2 *  5000000) - 1,
+    DIV_UP(CLOCK_CORECLOCK, 2 * 10000000) - 1,
+};
+#undef DIV_UP
+#define SPI_NUMOF                   (1)
```

Comment:
I would like to have some struct based configuration. See #12917 where I changed the UART to use this scheme. I find this convenient, at least to know which pins are connected the serial interface and it's more consistent with most of other board configurations in RIOT (like STM, SAM, nRF).